javacord-core/src/main/java/org/javacord/core/util/handler/channel/thread/ThreadUpdateHandler.java

```diff
@@ -52,22 +83,164 @@ private void handleThread(final JsonNode jsonChannel) {
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeArchivedEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final boolean wasLocked = metadata.isLocked();
+        final boolean isLocked = metadataJson.get("locked").asBoolean();
+        if (wasLocked != isLocked) {
+            metadata.setLocked(isLocked);
+
+            final ServerThreadChannelChangeLockedEvent event =
+                    new ServerThreadChannelChangeLockedEventImpl(thread, isLocked, wasLocked);
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeLockedEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final Instant oldArchiveTimestamp = metadata.getArchiveTimestamp();
+        final Instant newArchiveTimestamp = OffsetDateTime.parse(metadataJson.get("archive_timestamp")
+                .asText()).toInstant();
+        if (!Objects.deepEquals(oldArchiveTimestamp, newArchiveTimestamp)) {
+            metadata.setArchiveTimestamp(newArchiveTimestamp);
+
+            final ServerThreadChannelChangeArchiveTimestampEvent event =
+                    new ServerThreadChannelChangeArchiveTimestampEventImpl(thread,
+                            newArchiveTimestamp, oldArchiveTimestamp);
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeArchiveTimestampEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final Optional<Boolean> wasInvitable = metadata.isInvitable();
+        final Optional<Boolean> isInvitable = Optional.ofNullable(metadataJson.hasNonNull("invitable")
+                ? metadataJson.get("invitable").asBoolean() : null);
+
+        if (isInvitable.isPresent() && wasInvitable.isPresent()) {
+            metadata.setInvitable(isInvitable.get());
+
+            final ServerThreadChannelChangeInvitableEvent event =
+                    new ServerThreadChannelChangeInvitableEventImpl(thread, isInvitable.get(), wasInvitable.get());
+
+            api.getEventDispatcher().dispatchServerThreadChannelChangeInvitableEvent(
+                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
+        }
+
+        final Set<ThreadMember> oldMembers = thread.getMembers();
+        final Set<ThreadMember> newMembers = new HashSet<>();
+        if (jsonChannel.hasNonNull("member")) {
+            final Set<ThreadMember> members = thread.getMembers();
+            final ThreadMember member = new ThreadMemberImpl(api, server, jsonChannel.get("member"));
+            members.add(member);
+            thread.setMembers(newMembers);
+            final ServerPrivateThreadJoinEvent event =
+                    new ServerPrivateThreadJoinEventImpl(thread, newMembers,
+                            oldMembers);
```

Comment:
```suggestion
                    new ServerPrivateThreadJoinEventImpl(thread, member);
```
You have still not removed the members from the event...
Instead just pass the new ThreadMember of the bot

Reply:
will do so

Reply:
will this work 
```java
  if (jsonChannel.hasNonNull("member")) {
            final Set<ThreadMember> members = new HashSet<>(thread.getMembers());
            final ThreadMember member = new ThreadMemberImpl(api, server, jsonChannel.get("member"));
            members.add(member);
            thread.setMembers(members);

            final ServerPrivateThreadJoinEvent event =
                    new ServerPrivateThreadJoinEventImpl(thread, member);

            api.getEventDispatcher().dispatchServerPrivateThreadJoinEvent(
                    (DispatchQueueSelector) thread.getServer(), thread.getServer(), thread, event);
        }
```